src/arch/io/disk.cc

```diff
@@ -623,6 +629,11 @@ int perform_datasync(fd_t fd) {
     int res = fdatasync(fd);
     return res == -1 ? get_errno() : 0;
 
+#elif defined(__FreeBSD__)
```

Comment:
We should be calling fdatasync here, like with linux, no?

Reply:
> The problem is that FreeBSD doesn't have a fdatasync(2) system call :(

https://www.freebsd.org/cgi/man.cgi?query=fdatasync&sektion=2

Reply:
And now it does :eyes: